Allow custom 404.html on projects (#5130)
rtfd/readthedocs.org: docs/guides/custom-404-page.rst

Use custom 404 page on my project
=================================

If you want your project to use a custom page for not found pages instead of the "Maze Found" default one,
you can put a ``404.html`` on the root of your project's output.

> Once a 404 is about to be returned, Read the Docs first checks if there is a ``404.html`` in the root of your project's output and if it does exist, it's returned.

Reviewer: ```suggestion
When a 404 is returned, Read the Docs checks if there is a ``404.html`` in the root of your project's output and uses it if it exists.
```

As the ``404.html`` will be returned for all the URLs where the real page was not found,

> all its resources URLs and links must be absolute (including internals),

Reviewer: ```suggestion
all its resources URLs and links must be absolute (start with a `/`),
```

> otherwise they won't be found by the client for all the 404 URLs producing errors while displaying the page (styles and/or images not found).

Reviewer: ```suggestion
otherwise they will not work when a user clicks on them.
```

In case you want to follow the same style for the 404 page than your theme, you can either:

> 1. manually copy the source of any already rendered pages or,

Reviewer: We shouldn't recommend this, but just point to the extension.

Author: I removed this item and the whole "Manually creation" section.
